index.js

```diff
@@ -9,17 +9,14 @@ const create = prod => (start, sink) => {
   let got2;
   let unsub;
   sink(0, t => {
-    got2 = got2 || t === 2;
+    end = end || t === 2;
+    end && unsub && unsub();
```

Comment:
just because of linter issue please use this form `if (end && unsub) unsub();`

Reply:
fixed